contracts/USM.sol

```diff
@@ -44,52 +57,24 @@ contract USM is ERC20 {
      * @param _usmAmount Amount of USM to burn.
      */
     function burn(uint _usmAmount) external {
-        uint ethAmount = _applyFee(_usmToEth(_usmAmount), BURN_FEE);
-        ethPool = ethPool.sub(ethAmount);
+        uint ethAmount = _usmToEth(_usmAmount);
+        uint ethMinusFee = ethAmount.sub(muld(ethAmount, BURN_FEE));
+        ethPool = ethPool.sub(ethMinusFee);
         _burn(msg.sender, _usmAmount);
-        Address.sendValue(msg.sender, ethAmount);
+        Address.sendValue(msg.sender, ethMinusFee);
     }
 
     /**
      * @notice Calculate debt ratio of the current Eth pool amount and outstanding USM
      * (the amount of USM in total supply).
      *
-     * @return Debt ratio decimal percentage *(10**decimals()). For example, 100%
-     * is denoted as 1,000,000,000,000,000,000 (1 with 18 zeros).
+     * @return Debt ratio in UNIT.
      */
     function debtRatio() external view returns (uint) {
-        return _debtRatio(ethPool, totalSupply());
-    }
-
-    /**
-     * @notice Calculate debt ratio of an Eth pool amount and outstanding USM
-     * (the amount of USM in total supply).
-     *
-     * @param _ethPool Amount of Eth in the pool.
-     * @param _usmOutstanding Amount of USM in total supply.
-     * @return Debt ratio decimal percentage *(10**decimals()). For example, 100%
-     * is denoted as 1,000,000,000,000,000,000 (1 with 18 zeros).
-     */
-    function _debtRatio(uint _ethPool, uint _usmOutstanding) internal view returns (uint) {
-        if (_ethPool == 0) {
+        if (ethPool == 0) {
             return 0;
         }
-        return _usmOutstanding.mul(10**uint(decimals())).div(_ethToUsm(_ethPool));
-    }
-
-    /**
-     * @notice Apply fees to a number, the result being the number, minus the
-     * percentage fee.
-     *
-     * @param _number The amount that the fee is being applied to
-     * @param _feePercentage Percentage fee to be subtracted from the number
-     * in parts per 10,000.
-     * @return The result of the _number minus the percentage fee.
-     */
-    function _applyFee(uint _number, uint _feePercentage) internal pure returns (uint) {
-        uint result = _number.sub((_number.mul(_feePercentage)).div(PERCENTAGE_BASE));
-        require(result < _number, "Fee must make the result smaller");
```

Comment:
These `require` are not needed, because both parameters are `uint` and therefore positive, and `.sub` protects us against overflow.